app/views/insured/plan_shoppings/show.html.erb

```diff
@@ -1,635 +1,1173 @@
-            }
+      function showFilteredResults() {
+        var element = document.getElementById('filteredPlans');
+        filteredArray.forEach(function (result) {
+          var logo = carrierLogos[result.product.issuer_name];
+          if (result.product.kind == 'health') {
+            var plan_kind = result.product.health_plan_kind
           }
-          if (networkType.toLowerCase() == 'nationwide') {
-            if (enrollment.product.nationwide) {
-              resultsData.push(enrollment);
-            }
+          else if (result.product.dental_plan_kind) {
+            var plan_kind = result.product.dental_plan_kind;
+          }
+          // Used to show deductible for plan
+          if (result.member_enrollments.length > 1) {
+            var deductible = result.product.family_deductible;
+            var deductibeType = "FAMILY DEDUCTIBLE";
+          } else {
+            var deductible = result.product.deductible;
+            var deductibeType = "DEDUCTIBLE";
+          }
+          if (result.product.sbc_document) {
+            var sbc_url = '/document/download/'
+            + result.product.sbc_document.identifier.split("#")[0].split(":").pop() + '/'
+            + result.product.sbc_document.identifier.split("#")[1] + '?content_type=application/pdf&amp;filename='
+            + result.product.title.replace(/[^0-9a-z]/gi, '') + '.pdf&amp;disposition=inline';
+          }
+          else {
+            var sbc_url = "";
           }
-        })
-      })
-      return resultsData;
-    }
 
-    function filterResultsSelections(selections) {
-      var resultsArray;
-      var filtered = false;
+          var network = "";
+          if (result.product.dc_in_network) {
+            network = "<%= EnrollRegistry[:enroll_app].setting(:statewide_area).item %>";
+          }
 
-      metalLevels = selections.selectedMetalLevels;
-      planTypes = selections.selectedPlanTypes;
-      networkTypes = selections.selectedPlanNetworks;
-      carrier = selections.selectedCarrier;
-      selectedHSA = selections.selectedHSA;
-      selectedOSSE = selections.selectedOSSE;
-      resultsArray = availableEnrollments;
+          if (result.product.nationwide) {
+            network = "<%= EnrollRegistry[:enroll_app].setting(:nationwide_area).item %>";
+          }
 
-      if (metalLevels.length) {
-        filtered = true;
-        resultsArray = filterByMetalLevels(metalLevels, resultsArray)
-      }
+          var standard_plan = "";
+          if (result.product.is_standard_plan) {
+            standard_plan = "<%= EnrollRegistry[:enroll_app].setting(:standard_plan_label).item %>";
+          }
 
-      if (planTypes.length) {
-        filtered = true;
-        resultsArray = filterByPlanTypes(planTypes, resultsArray)
+          var div = document.createElement('div');
+          div.setAttribute('class', 'myfilteredPlans');
+          div.innerHTML =
+          `
+          <div class="module plan-row panel panel-default">
```

Comment:
there's a lot of HTML here. is it worth breaking this out into a partial and doing something like `div.replaceWith("<%= escape_javascript render(partial: '...'}) %>")`